src/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useContext, useState } from 'react';
+
+import { TodosContext } from './TodosContext';
+import { TodoList } from '../components/TodoList/TodoList';
+import { TodosFilter } from '../components/TodosFilter/TodoFilter';
+import { Filter } from '../types/Filter';
+
+type Props = {
+};
+
+export const TodoApp: React.FC<Props> = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const [title, setTitle] = useState('');
+  const [filter, setFilter] = useState(Filter.ALL);
+
+  const handlerFilterChange = (newFilter: string) => {
+    setFilter(newFilter);
+  };
+
+  const handlerChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handlerAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (title.trim() !== '') {
+      const newTask = {
+        id: +new Date(),
+        title,
+        completed: false,
+      };
+
+      setTodos((prevTodos) => [...prevTodos, newTask]);
+      setTitle('');
+    }
+  };
+
+  const handlerCompleteAll = () => {
+    if (todos.some(todo => todo.completed === false)) {
+      const updatedTodos = todos.map((todo) => ({
+        ...todo,
+        completed: true,
+      }));
+
+      setTodos(updatedTodos);
+    } else {
+      const updatedTodos = todos.map((todo) => ({
+        ...todo,
+        completed: false,
+      }));
+
+      setTodos(updatedTodos);
+    }
+  };
+
+  const handlerClearCompletes = () => {
+    setTodos(currentTodos => currentTodos
+      .filter(elem => !elem.completed));
+  };
+
+  const filteredTodos = todos.filter((todo) => {
+    if (filter === Filter.ALL) {
+      return true;
+    }
+
+    if (filter === Filter.ACTIVE) {
+      return !todo.completed;
+    }
+
+    return todo.completed;
+  });
+
+  const noCompleteTodos = todos.filter(elem => !elem.completed);
+  const isSomeComplete = todos.some(todo => todo.completed === true);
+  const allCompleted = todos.every(todo => todo.completed === true);
```

Comment:
```suggestion
  const isSomeComplete = todos.some(todo => todo.completed);
  const allCompleted = todos.every(todo => todo.completed);
```
redundant to compare two boolean values

also looks like redundant to make additional loop for isSomeCoplete.
we can compare length of todos and length of noCompleteTodos and will know whether we have some completed todos.
```
const isSomeComplete = noCompleteTodos.length !== todos.length;
```